src/Compilers/CSharp/Portable/Symbols/MemberSymbolExtensions.cs

```diff
@@ -418,6 +418,23 @@ internal static bool IsPartialDefinition(this Symbol member)
             return sms?.IsPartialDefinition == true;
         }
 
+        internal static bool ContainsTupleNames(this Symbol member)
```

Comment:
Curious: did you consider using pattern matching here? 

``` c#
switch (member) 
{
  case MethodSymbol method:
  ...
  case PropertySymbol property:
}
```

Mostly asking to see why we are / aren't using newer features in our changes. Possible they're just not applicable here.